core/src/main/java/org/elasticsearch/index/query/AndQueryParser.java

```diff
@@ -72,21 +72,11 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 } else if (parseContext.isDeprecatedSetting(currentFieldName)) {
                     // skip
                 } else if (token == XContentParser.Token.START_ARRAY) {
-                    if ("filters".equals(currentFieldName)) {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
-                        }
-                    } else {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
+                    queriesFound = true;
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                        Query filter = parseContext.parseInnerFilter();
+                        if (filter != null) {
+                            queries.add(filter);
```

Comment:
btw. that was also the reason I didn't want to get stricter than necessary here at first, we could also leave everything as-is and I only remove this extra bit of code on the query refactoring branch.